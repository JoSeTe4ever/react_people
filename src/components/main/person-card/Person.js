import React from 'react';
import './Person.css'
import Checkbox from "@material-ui/core/Checkbox/Checkbox";

class Person extends React.Component {

    constructor(props) {
        super(props);
        this.state = {};
    };

    render() {
        return (
            <div className="card">
                <img src={person.picture} alt="Avatar" className={"dynamicAvatar"}/>
                <div className="container">
                    <h4><b>{person.name}</b></h4>
                    <p>{person.address}</p>
                    <Checkbox></Checkbox>
                </div>
            </div>
        )
    }
};

export default Person;