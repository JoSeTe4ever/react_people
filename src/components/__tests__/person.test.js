import React from 'react';
import {shallow} from 'enzyme';

import Person from '../main/person-card/Person';

describe('Person component', () => {

    it('should render correctly with specific props', () => {
        //Arrange
        const jopi = {
            "_id": "5c20f8aa7960361f6ecb3ed6",
            "index": 1,
            "greeting": "Hello, Jopi! You have 10 unread messages.",
            "favoriteFruit": "strawberry",
            "name": "Jopi Piqueras",
            "address": "Smin Streen 78373",
            "picture": "https://api.adorable.io/avatars/66/lacynguyen@nutralab.com"
        };
        const handleCheckboxChange = () => {
        };

        //Act
        const personWrapper = shallow(<Person person={jopi}
                                              key={jopi._id}
                                              handleCheckboxChange={handleCheckboxChange}/>);

        //Assert
        expect(personWrapper).toMatchSnapshot();
    });

    it('should render a person card containing the model information', () => {
        //Arrange
        const jopi = {
            "_id": "5c20f8aa7960361f6ecb3ed6",
            "index": 1,
            "greeting": "Hello, Jopi! You have 10 unread messages.",
            "favoriteFruit": "strawberry",
            "name": "Jopi Piqueras",
            "address": "Smin Streen 78373",
            "picture": "https://api.adorable.io/avatars/66/lacynguyen@nutralab.com"
        };
        const handleCheckboxChange = () => {
        };

        //Act
        const personWrapper = shallow(<Person person={jopi}
                                        key={jopi._id}
                                        handleCheckboxChange={handleCheckboxChange}/>);

        //Assert
        expect(personWrapper.find('h4')).toHaveLength(1);
        expect(personWrapper.find('h4').text()).toBe("Jopi Piqueras");
        expect(personWrapper.find('p').text()).toBe("Smin Streen 78373");
        expect(personWrapper.find('img')).toHaveLength(1);
        expect(personWrapper.find('img').prop("src")).toBe("https://api.adorable.io/avatars/66/lacynguyen@nutralab.com");
    });
});