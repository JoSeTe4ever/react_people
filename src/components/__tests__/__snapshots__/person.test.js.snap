// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person component should render correctly with specific props 1`] = `
<div
  className="card"
>
  <img
    alt="Avatar"
    className="dynamicAvatar"
    src="https://api.adorable.io/avatars/66/lacynguyen@nutralab.com"
  />
  <div
    className="container"
  >
    <h4>
      <b>
        Jopi Piqueras
      </b>
    </h4>
    <p>
      Smin Streen 78373
    </p>
    <WithStyles(Checkbox) />
  </div>
</div>
`;
